includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,49 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		$class_prefixes = [
+			'CalendarDay',
+			'CalendarMonth',
+			'DateInput',
+			'DateRangePicker',
+			'DayPicker',
+			'KeyboardShortcutRow',
+			'SingleDatePicker',
+		];
+
+		foreach ( $class_prefixes as $prefix ) {
+			if ( 0 === strpos( $class, $prefix ) ) {
```

Comment:
I'm working on making this into a `preg_match()`.

Reply:
Applied in https://github.com/ampproject/amp-wp/pull/4339/commits/bd9313d462aca486423006824ff6725beac8df0d